### PR TITLE
Directly retrieve servlet context form request

### LINKED ISSUE #24879: Retrieve servlet context directly from request in /metrics servlet
Related to : https://github.com/OpenLiberty/open-liberty/issues/24590

Investigation with Sessions team produced advice from them to retrieve the servletContext directly form the request instead of retrieving a session as the servlet does not use the session for any other purpose.

### PR DESCRIPTION
**Pull request by Channyboy**
#build
Fixes #24879

### CODE REVIEW

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_suZr0NPaEe2LN8swt3RA1A&tab=com.ibm.team.build.web.ui.internal.editors.result.test.TestProvider

Personal build which ran the individual monitor metric FATS 10 times passed as well.

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetricsRESTProxyServlet.java [hunk 1]
```diff
@@ -101,6 +99,8 @@ private synchronized void getAndSetRESTHandlerContainer(HttpServletRequest reque
                 throw new ServletException("OSGi service RESTHandlerContainer is not available.");
             } else {
                 REST_HANDLER_CONTAINER = ctxt.getService(ref);
+                // generate initial session metrics
+                request.getSession();
```
Comment (Channyboy) [inline dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetricsRESTProxyServlet.java:103]: Still need to do this to generate the presence of session metrics.

File: dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/microprofile/metrics50/rest/BaseMetricsRESTProxyServlet.java [hunk 2]
```diff
@@ -103,6 +100,8 @@ private synchronized void getAndSetRESTHandlerContainer(HttpServletRequest reque
                 throw new ServletException("OSGi service RESTHandlerContainer is not available.");
             } else {
                 REST_HANDLER_CONTAINER = ctxt.getService(ref);
+                // generate initial session metrics
+                request.getSession();
```
Comment (Channyboy) [inline dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/microprofile/metrics50/rest/BaseMetricsRESTProxyServlet.java:104]: Still need to do this to generate the presence of session metrics.

File: dev/com.ibm.ws.microprofile.metrics/src/com/ibm/ws/microprofile/metrics/BaseMetricsRESTProxyServlet.java [hunk 3]
```diff
@@ -101,6 +99,8 @@ private synchronized void getAndSetRESTHandlerContainer(HttpServletRequest reque
                 throw new ServletException("OSGi service RESTHandlerContainer is not available.");
             } else {
                 REST_HANDLER_CONTAINER = ctxt.getService(ref);
+                // generate initial session metrics
+                request.getSession();
```
Comment (Channyboy) [inline dev/com.ibm.ws.microprofile.metrics/src/com/ibm/ws/microprofile/metrics/BaseMetricsRESTProxyServlet.java:103]: Still need to do this to generate the presence of session metrics.

**Review by fmhwong — APPROVED** [review]
LGTM

**Comment by Channyboy** [comment]
#libby

**Pull request merged**